### ISSUE TITLE
Error saving data - M2M

### ISSUE THREAD

**Issue by jreid06** [directus/app#1159]
# Bug Report

## Steps to Reproduce
my use case
1.  create a user table
2.  create a skills table
3. create a user_skills junction
4. in user table create M2M for column "user_skills" 
5. create a user and add a skill

## Expected Behavior

1. saves the relationship and creates a new entry in skills table & user skills (updating user_id & skills_id columns)

## Actual Behavior

1. returns error "Cannot read property 'interface' of undefined"

## Other Context & Screenshots

![image](https://user-images.githubusercontent.com/19251469/49305027-fa4ec100-f4c5-11e8-9e93-6ef3e28fe940.png)

## Technical Details

- Device: Macbook pro
- OS:MacOS 10.14.1
- Browser: Chrome 68.0.3440.106
- Directus App:  latest build version of app
- Directus API: latest build branch version
- Install Method: build version of app

**Comment by iLama** [comment]
I came across this problem today. It seems like the bug is related to the schema of the related tables and the junction table.

**Comment by rijkvanzanten** [comment]
@jreid06 I have some trouble reproducing it. The only thing I noticed is that I had to refresh the app once before the relationship could be found, but that gave me another issue then the one you're describing. I'm thinking you might have made a little mistake in the setup of the many to many interface. I'm (successfully) using:

<img width="811" alt="image" src="https://user-images.githubusercontent.com/9141017/49318177-84e2f080-f4c5-11e8-97be-85af89ee5afe.png">

<img width="737" alt="image" src="https://user-images.githubusercontent.com/9141017/49318261-d4c1b780-f4c5-11e8-85a3-aa0c7ab66b20.png">
